common/setups/rasr/nn_system.py

```diff
@@ -0,0 +1,554 @@
+
+    def generate_lattices(self):
+        pass
+
+    # -------------------- Training --------------------
+
+    def returnn_training(
+        self,
+        name,
+        returnn_config,
+        nn_train_args,
+        train_corpus_key,
+        cv_corpus_key,
+        devtrain_corpus_key=None,
+    ):
+        assert isinstance(returnn_config, returnn.ReturnnConfig)
+
+        returnn_config.config["train"] = self.train_input_data[
+            train_corpus_key
+        ].get_data_dict()
+        returnn_config.config["dev"] = self.cv_input_data[cv_corpus_key].get_data_dict()
+        if devtrain_corpus_key is not None:
+            returnn_config.config["eval_datasets"] = {
+                "devtrain": self.devtrain_input_data[
+                    devtrain_corpus_key
+                ].get_data_dict()
+            }
+
+        train_job = returnn.ReturnnTrainingJob(
+            returnn_config=returnn_config,
+            **nn_train_args,
+        )
+
+        train_job.add_alias(f"train_nn/{train_corpus_key}_{cv_corpus_key}/{name}_train")
+        self.jobs[f"{train_corpus_key}_{cv_corpus_key}"][name] = train_job
+        self.nn_models[f"{train_corpus_key}_{cv_corpus_key}"][
+            name
+        ] = train_job.out_models
+        self.nn_configs[f"{train_corpus_key}_{cv_corpus_key}"][
+            name
+        ] = train_job.out_returnn_config_file
+
+        return train_job
+
+    def returnn_rasr_training(
+        self,
+        name,
+        returnn_config,
+        nn_train_args,
+        train_corpus_key,
+        cv_corpus_key,
+    ):
+        train_data = self.train_input_data[train_corpus_key]
+        dev_data = self.cv_input_data[cv_corpus_key]
+
+        train_crp = train_data.get_crp()
+        dev_crp = dev_data.get_crp()
+
+        assert train_data.features == dev_data.features
+        assert train_data.alignments == dev_data.alignments
+
+        assert isinstance(returnn_config, returnn.ReturnnConfig)
+
+        train_job = returnn.ReturnnRasrTrainingJob(
+            train_crp=train_crp,
+            dev_crp=dev_crp,
+            feature_flow=train_data.feature_flow,
+            alignment=train_data.alignments,
+            returnn_config=returnn_config,
+            **nn_train_args,
+        )
+
+        train_job.add_alias(f"train_nn/{train_corpus_key}_{cv_corpus_key}/{name}_train")
+        self.jobs[f"{train_corpus_key}_{cv_corpus_key}"][name] = train_job
+        self.nn_models[f"{train_corpus_key}_{cv_corpus_key}"][
+            name
+        ] = train_job.out_models
+        self.nn_configs[f"{train_corpus_key}_{cv_corpus_key}"][
+            name
+        ] = train_job.out_returnn_config_file
+
+        return train_job
+
+    # -------------------- Recognition --------------------
+
+    def nn_recognition(
+        self,
+        name: str,
+        returnn_config: returnn.ReturnnConfig,
+        checkpoints: List[returnn.Checkpoint],
+        acoustic_mixture_path: tk.Path,
```

Comment:
`acoustic_mixture_path` could be optional, if a prior file is provided.
(ok, not really.. we would still need a dummy mixture)